packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1492,13 +1498,23 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   // frame so that we minimize the amount of building that happens.
   Widget? _modalScopeCache;
 
+  void _onMountScope() {
```

Comment:
Doesn't this problem (where a widget accesses properties of a route) also affect routes that are not ModalRoutes?

Reply:
Only `ModalRoute` and its subclasses are aware of widgets.